src/Functions/FunctionsConversion.h

```diff
@@ -683,12 +722,23 @@ struct ConvertImpl<FromDataType, std::enable_if_t<!std::is_same_v<FromDataType,
 
             for (size_t i = 0; i < size; ++i)
             {
-                FormatImpl<FromDataType>::execute(vec_from[i], write_buffer, &type, time_zone);
+                if (null_map)
```

Comment:
@vdimir it seems that because of this check performance for toString for non nullable type is affected. My proposal is to move this check if out of hot loop.
And check if that helps.
```c++
if (null_map)
{
    for (size_t i = 0; i < size; ++i)
    {
         bool is_ok = FormatImpl<FromDataType>::template execute<bool>(vec_from[i], write_buffer, &type, time_zone);
         null_map->getData()[i] |= !is_ok;
    }
}
else
{
    for (size_t i = 0; i < size; ++i)
    {
          FormatImpl<FromDataType>::template execute<void>(vec_from[i], write_buffer, &type, time_zone);
    }
}
```